### PR TITLE
[Gradient Compression] Refactor tensor grouping in PowerSGD

### PR DESCRIPTION
**Pull request by wayi1**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #53099 [Gradient Compression] Make GradBucket class public
* **#52981 [Gradient Compression] Refactor tensor grouping in PowerSGD**

No need to create a hard boundary between rank-1 tensors and high-rank tensors, since some high-rank tensors will not be compressed if the compression cannot save enough bandwidth, according to `_should_compress` function.

Therefore, refactor and simplify the tensor grouping logic, which addresses the comment in https://github.com/pytorch/pytorch/pull/52541#discussion_r580867311

Differential Revision: [D26713689](https://our.internmc.facebook.com/intern/diff/D26713689/)

### CODE REVIEW

**Comment by SciPioneer** [comment]
@tvogels Can you take a look?

Particularly, I used `n, m = tensor.shape[0], tensor.numel() / tensor.shape[0]` instead of
`n, m = tensor.shape[0], tensor.nelement() // tensor.shape[0]` in your [comment](https://github.com/pytorch/pytorch/pull/52541#discussion_r580867311).

**Comment by tvogels** [comment]
Looks good to me.

File: torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py [hunk 1]
```diff
@@ -180,36 +180,34 @@ def powerSGD_hook(
     Once gradient tensors are aggregated across all workers, this hook applies
     compression as follows:
 
-    1. Views the input flattened 1D gradient tensor as two groups of per-parameter tensors: high-rank tensors and vector-like rank-1 tensors (for biases).
+    1. Views the input flattened 1D gradient tensor as a list of per-parameter tensors, and divides all the tensors into two groups:
 
-    2. Divides all the tensors into two groups:
+        1.1 The tensors that should be compressed before allreduce, because the compression can give enough saving in bandwidth.
```
Comment (rohan-varma) [inline torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py:185]: Do we specify somewhere how we determine which tensors are compressed vs uncompressed (rank 1 & the compression threshold?)

**Review by rohan-varma — APPROVED** [review]
LGTM! Thanks for refactoring this

File: torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py (hunk 1 above)
Comment (SciPioneer) [reply torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py:185]: It's specified in `_should_compress` method, noted in the precursor PR.

**Pull request closed without merging**